components/ui/VersionFilterControl.vue

```diff
@@ -95,7 +95,29 @@ export default {
       selectedLoaders: [],
     }
   },
+  watch: {
+    showSnapshots() {
+      this.updateQuery()
+    },
+  },
+  mounted() {
+    this.selectedLoaders = this.getQueryAsArray(this.$route.query.loaders)
+    this.selectedGameVersions = this.getQueryAsArray(
+      this.$route.query.gameVersions
+    )
+    this.showSnapshots = this.$route.query.showSnapshots === 'true'
+    this.updateVersionFilters()
+  },
   methods: {
+    getQueryAsArray(query) {
+      if (Array.isArray(query)) {
+        return query
+      } else if (typeof query === 'string') {
+        return query.split(',')
+      } else {
+        return []
+      }
```

Comment:
is there a reason why both of these cases are supported? IMO for simplicity the version filters should always be in one format. I think just keeping it as a JS array is best (since this is what is done when it is serialized)

Reply:
The query string is only represented as an array if there are multiple query parameters with the same key:

**This returns an array**
`loaders=forge&loaders=fabric`

**This returns a string**
`loaders=forge,fabric`

So we have to handle both cases, since both of them are possible.

Reply:
> The query string is only represented as an array if there are multiple query parameters with the same key:
> 
> **This returns an array** `loaders=forge&loaders=fabric`
> 
> **This returns a string** `loaders=forge,fabric`
> 
> So we have to handle both cases, since both of them are possible.

I'm confused how they are both possible. That doesn't make sense. The UI is what creates the URLs which are shared with people. Does the UI create both of these instances, if so, why? If it doesn't, then the code that handles what the UI doesn't "create" in the URL should be removed. We should only handle supported cases in URL params. Otherwise it introduces complexity which isn't needed. (see how search doesn't need this)

Reply:
The array case is only if the user creates the URL manually with the same query parameter multiple times. But that case still needs to be handled, since it would lead to an error otherwise.

Reply:
>  only if the user creates the URL manually 

Users should not be doing this and we should not support this case